framework/src/systems/NonlinearSystem.C

```diff
@@ -141,6 +142,47 @@ NonlinearSystem::addMatrix(TagID tag)
   return *mat;
 }
 
+void
+NonlinearSystem::applyPredictor(NumericVector<Number> & initial_solution)
+{
+  if (_fe_problem.solverParams()._type != Moose::ST_LINEAR)
+  {
+    // retain a copy of the unpredicted solution
+    auto unpredicted_solution = initial_solution.clone();
+
+    _predictor->apply(initial_solution);
+    _fe_problem.predictorCleanup(initial_solution);
+
+    // compute the residual of the prediction
+    _sys.solution->close();
```

Comment:
Do you get an error message if you do not close solution? I am asking because `close` is not a cheap operation in parallel. We try to avoid this operation if possible.

Reply:
Let me recheck. I did not get an error, but I thought I got the wrong results...